robe.el

```diff
@@ -83,6 +83,25 @@ have constants, methods and arguments highlighted in color."
 
 (defvar robe-running nil)
 
+(defcustom robe-completing-read-func 'ido-completing-read
+  "Function to call for completing read. Will not be used when either
```

Comment:
Commit summary: done. Will push --force momentarily. 

Default: done. I'd started with the revert, which had the ido default.

Docstring: done.

completing-read can be overwritten with a `flet` type of construct. Company-quickhelp does exactly this, and we needed to fix that interaction:  https://github.com/expez/company-quickhelp/blob/master/company-quickhelp.el#L115

You can argue that `let (c-r-function` would have been better, but this is not uncommon. Or it could be a user `defun`.